tyr/servers/server.py

```diff
@@ -597,74 +620,88 @@ def resolve_iam_role(self):
 
         self.IAM_ROLE_POLICIES.extend(self.GLOBAL_IAM_ROLE_POLICIES)
         self.IAM_ROLE_POLICIES = list(set(self.IAM_ROLE_POLICIES))
-        for policy_template in self.IAM_ROLE_POLICIES:
-            policy = policy_template.format(environment=self.environment)
-
-            self.log.info('Processing policy "{policy}"'.format(policy=policy))
-
-            if policy not in existing_policies:
 
-                rolePolicy = policies[policy]
+        # If managed policies exist, then use these instead of line policies:
+        if (len(self.IAM_MANAGED_POLICIES) > 0):
+            self.log.info("Adding managed policies [" + str(self.IAM_MANAGED_POLICIES).format(environment=self.environment) + "] to role [" + self.role + "]")
 
-                if rolePolicy is None:
-                    self.log.info("No policy defined for {policy}".format(
-                                  policy=policy))
-                    continue  # Go to the next policy
+            for m_policy in self.IAM_MANAGED_POLICIES:
+                m_policy_id = m_policy.format(environment=self.environment)
+                arn = self.iam.get_user().user.arn
+                account_id = arn[arn.find('::')+2:arn.rfind(':')]
+                m_policy_arn = self.iam.get_policy("arn:aws:iam::{account_id}:policy/{policy}".format(account_id=account_id,policy=m_policy_id))
+                self.iam.attach_role_policy("arn:aws:iam::{account_id}:policy/{policy}".format(account_id=account_id, policy=m_policy_id), self.role)
 
-                self.log.info('Policy "{policy}" does not exist'.format(
-                              policy=policy))
+        # Use inline roles instead of managed policies:
+        else:
```

Comment:
Hey mate, my understanding of the Jira was the server should either be setup with managed policies or inline policies but not both, I can correct this if I am not right on this